core-bundle/src/Resources/contao/widgets/SerpPreview.php

```diff
@@ -128,13 +128,14 @@ private function getUrl(Model $model)
 			throw new \LogicException('No url_callback given');
 		}
 
-		$alias = $this->getAlias($model);
+		$aliasField = $this->aliasField ?: 'alias';
 		$placeholder = bin2hex(random_bytes(10));
 
 		// Pass a detached clone with the alias set to the placeholder
 		$tempModel = clone $model;
-		$tempModel->origAlias = $tempModel->$alias;
-		$tempModel->$alias = $placeholder;
+		$tempModel->{$tempModel::getPk()} = $model->{$model::getPk()};
```

Comment:
This line should not be necessary according to the initial issue description.

> If the model has **no ID and no alias** this call will throw an exception.

Reply:
It is necessary. Because if the alias is empty, `PageModel::getAbsoluteUrl()` will use the ID, and if that is empty as well, we'll get the exception.

Reply:
I see. Then we should use `cloneOriginal()` instead.

```php
		$aliasField = $this->aliasField ?: 'alias';
		$placeholder = bin2hex(random_bytes(10));

		// Pass a detached clone with the alias set to the placeholder
		$tempModel = $model->cloneOriginal();
		$tempModel->origAlias = $tempModel->$aliasField;
		$tempModel->$aliasField = $placeholder;
		$tempModel->preventSaving(false);
```